src/main/scala/scatan/model/game/ops/CardOps.scala

```diff
@@ -10,6 +10,33 @@ object CardOps:
 
   extension (state: ScatanState)
 
+    /** Stole a card from a victim.
+      * @param currentPlayer
+      *   the player who stole the card
+      * @param victim
+      *   the player who lost the card
+      * @return
+      *   Some(ScatanState) if the card was stolen, None otherwise
+      */
+    def stoleResourceCard(currentPlayer: ScatanPlayer, victim: ScatanPlayer): Option[ScatanState] =
+      val victimResourceCards = state.resourceCards(victim)
+      if victimResourceCards.sizeIs == 0 then None
+      else
+        val randomCardIndex = scala.util.Random.nextInt(victimResourceCards.size)
+        val stolenCard = victimResourceCards(randomCardIndex)
+        val updatedVictimResourceCards = victimResourceCards.filterNot(_ == stolenCard)
```

Comment:
Are you sure that this `filterNot` doesn't take all the cards of the same type of **stolenCard** instead of taking only that one?